### PR TITLE
new "mode" flag

### PR DESCRIPTION
**Pull request by indaco**
This PR:

- bump `templ` from _v0.2.731_ to the latest _v0.2.747_
- Add a new `m` flag to address the use-case of generating the HTML file close to the respective template files instead of in an output directory. The possible values are `bundle` and `inline`. `bundle` is the **default** value and reflects the original `static-templ` way of working.
  - refactor `main` to mantain the same cyclomatic complexity as before
  - edit `generator.go` to include `Generate` function for each op mode
- update `README.md` file to include a description for the operational modes

### CODE REVIEW

**Comment by indaco** [comment]
@nokacper24 with this I think I am set for now :)

Thank you for your availability to support my use case, which allowed me to add new features and perform refactoring. 
I hope `static-templ` is improving with my contributions.

If this works for you, I think you could release `v1.1.0` after merging it.

File: main.go
Comment (nokacper24) [inline main.go]: I know I approved thin in previous PR. However, later I have noticed that if no arguments are passed the usage is  printed and the tool quits. Technically this is not okay, since there are default values provided for all flags, thus the tool should run just fine.

File: main.go
Comment (nokacper24) [inline main.go]: this ties in with the comment above, if the `len(os.Args) < 2` requirement is removed this will panic with index out of range.

File: main.go
Comment (nokacper24) [inline main.go]: this helper function results in a nil pointer dereference if no `-g` flag is provided, since it return nil if the flag is false. Replace these helper fucntions with simple if statements here, instead of run...ifNeeded

I'm assuming you are trying to keep the cyclomatic complexity low, but I'd argue it's better to keep the logic simple and "inline", rather than moving it to many helpers and potentially introduce bugs in the process.

Also, I believe the call to `runTemplGenerate` could be done before `finder.FindFilesInDir` (which returns the `groupedFiles`), this way you don't end up calling `finder.FindFilesInDir` twice.

**Review by nokacper24 — CHANGES_REQUESTED** [review]
Thank you! I found a few issues.
I can merge this in if you address my comments :)

**Comment by indaco** [comment]
Will review your comments and work on them

File: main.go
Comment (indaco) [reply main.go]: Ok

File: main.go
Comment (indaco) [reply main.go]: My goal is not keep the cyclomatic complexity low forcing design :) will do some rework

**Comment by indaco** [comment]
@nokacper24 

[b012a7b](https://github.com/nokacper24/static-templ/commit/b012a7bea3fdb07c6bd1e750f557296e81c896fb)

should address your comments, fixing the "run with defaults" as well as "keep the logic simple and inline".

I revert the code design for `main.go` as it was before this PR, adding the new  `m` flag and a `switch...case` statement on `flags.Mode`. 

Please, review it. Thanks

**Comment by nokacper24** [comment]
Thank you for contributing! Will do a new release in a moment.

**Comment by indaco** [comment]
@nokacper24 thank you for accepting this

The autogenerated release notes looks great :)

**Pull request merged**
